packages/core/src/openapi/openapi-request-builder.ts

```diff
@@ -1,42 +1,35 @@
+/* eslint-disable max-classes-per-file */
 import { AxiosResponse } from 'axios';
-import { Destination, DestinationNameAndJwt } from '../connectivity/scp-cf';
-import { buildAxiosRequestConfig } from '../http-client';
-import type {
-  ConstructorType,
-  FunctionReturnType,
-  ParametersType,
-  UnwrapAxiosResponse
-} from './types';
+import { Destination, DestinationNameAndJwt } from '../connectivity';
+import { executeHttpRequest, HttpResponse, Method } from '../http-client';
 
 /**
- * @experimental This API is experimental and might change in newer versions. Use with caution.
- * Generic request builder for building and executing requests based on apis generated by the OpenAPI generator.
- * @typeparam ApiT Interface of the generated api. All properties of the interface will be treated as functions.
- * @typeparam FnT Literal string type of the function name, must be a key of `ApiT` and represent a function.
+ * Request builder for OpenAPI requests.
+ * @typeParam ResponseT Type of the response for the request.
  */
-export class OpenApiRequestBuilder<ApiT, FnT extends keyof ApiT> {
+export class OpenApiRequestBuilder<ResponseT = any> {
+  private static isPlaceholder(pathPart: string): boolean {
+    return /^\{.+\}$/.test(pathPart);
+  }
+
   private customHeaders: Record<string, string> = {};
-  private args: ParametersType<ApiT, FnT>;
 
   /**
-   * Create an instance of `RestRequestBuilder`.
-   * @param apiConstructor Constructor of the underlying OpenApi api definition.
-   * @param fn Name of the function represented in thie request builder.
-   * @param args Arguments to pass to the api function.
+   * Create an instance of `OpenApiRequestBuilder`.
+   * @param method HTTP method of the request to be built.
+   * @param pathPattern Path for the request containing path parameter references as in the OpenAPI specification.
+   * @param parameters Query parameters and or body to pass to the request.
    */
   constructor(
-    private apiConstructor: ConstructorType<ApiT>,
-    public fn: FnT,
-    ...args: ParametersType<ApiT, FnT>
-  ) {
-    this.args = args;
-  }
+    public method: Method,
+    private pathPattern: string,
```

Comment:
I think you wanted to change this to a list of parameters and the path building is done in the builder, or is this a follow up?

Reply:
No I did it, that is why it is not the path pattern again, not the path template.